lib/waterline/query/dql/update.js

```diff
@@ -137,6 +137,7 @@ function createBelongsTo(valuesObject, cb) {
       var pk = val[model.primaryKey];
 
       valuesObject.values[item] = pk;
+      _.remove(valuesObject.associations.models, function(_item) { return _item == item; });
```

Comment:
Perhaps this could be simpler and have a comment?
```javascript
// now we have pk value attached, remove it from models
delete valuesObject.associations.models[item];
```